File: uts-java-1/Car.java
import java.util.Scanner;

public class Car extends Vehicle {
    private Wheel[] wheels;

    public void setWheels(int wheels){
        if (wheels == 0){
            wheels = 4;
            this.wheels = new Wheel[wheels];
            for (int i = 0; i < wheels; i++){
                this.wheels[i] = new Wheel(); //membuat objek wheel dengan default value
            }
            return;
        } else if(wheels < 3){
            Scanner sc = new Scanner(System.in);

            this.wheels = new Wheel[3];
            int index = 0;
            float harga = 0;
            float rata = 0;

            for (int i = 0; i < wheels; i++){  // Masukan price untuk jumlah wheel yang user input dahulu
                this.wheels[i] = new Wheel();
                System.out.print("Masukan harga wheel ke-" + String.valueOf(i + 1) + " : ");
                int t = Integer.parseInt(sc.next());
                this.wheels[i].setPrice(t);
                index++;
                harga += t;
            }
            sc.close();

            rata = harga / (index+1);
            // set harga tambahan
            int j = wheels;
            while (wheels < 3) {
                this.wheels[index] = new Wheel();
                harga += Math.round(rata);
                this.wheels[index].setPrice(Math.round(rata)); //bulatkan karena dibutuhkan int
                rata = harga / (index+1);
                wheels++;
            }
            
            System.out.println("\nAnda memasukan kurang dari 3 wheels, akan menambahkan " + String.valueOf(3 - j) + " wheels dengan harga rata\" dari wheels sebelumnya.");
        }
    }

    public int getPrice(){
        int sum = 0;
        for (Wheel wheel : wheels) {
            sum += wheel.getPrice();
        }

        return super.getBasePrice() + sum;
    }
}

File: Car.py
from Vehicle import Vehicle
from Wheel import Wheel

class Car(Vehicle):
    def __init__(self):
        super().__init__()
        self.__wheels = []
    
    def getPrice(self) -> int:
        wheel_price = 0
        for p in self.__wheels:
            wheel_price = wheel_price + p.price
        return super().basePrice + wheel_price
    
    def setWheels(self, wheels):
        # Kalau jumlah objek roda >= 3 maka wheels tidak perlu ditambahkan apa-apa
        if len(wheels) >= 3:
            self.__wheels = wheels
        
        # Kalau jumlah objek roda = 0 maka wheels akan ditambahkan sebanyak 4 wheel
        elif len(wheels) == 0:
            for c in range(4):
                w = Wheel() # sudah rp.1000
                self.__wheels.append(w)

        # Kalau jumlah objek roda = 1 maka wheels akan ditambahkan sebanyak 2 wheel
        elif len(wheels) == 1:
            self.__wheels = wheels # untuk meng-append yang diinput user
            for c in range(2):
                w = Wheel()
                w.price = wheels[0].price #karna cuman 1 buah roda jadi tidak cari rata-rata
                self.__wheels.append(w)

        # Kalau jumlah objek roda = 2 maka wheels akan ditambahkan sebanyak 1 wheel
        elif len(wheels) == 2:
            self.__wheels = wheels # untuk meng-append yang diinput user
            w = Wheel()
            w.price = (wheels[0].price + wheels[1].price)/2 #rata-rata wheels keduanya
            self.__wheels.append(w)